"This regexp should always match" exception

Not yet sure if that is my fault, or dependencies, or pyout -- exception message gives me no clues on what went wrong:
```shell
(git-annex)lena:~/proj/dandi/nwb-datasets/nwb_test_data[master]
$> dandi --pdb upload -i local -c nwb_test_data -t . v2.0.0/test_Clustering.nwb
--------------------------------------------------------------------------
[[29421,1],0]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)
  Host: lena

Another transport will be used instead, although this may result in
lower performance.

NOTE: You can disable this warning by setting the MCA parameter
btl_base_warn_component_unused to 0.
--------------------------------------------------------------------------
PATH                       SIZE    ERRORS UPLOAD STATUS                MESSAGE
v2.0.0/test_Clustering.nwb 17.8 kB   0           uploading                    


ERROR: 1 asynchronous worker failed

Producing value for row ('v2.0.0/test_Clustering.nwb',) failed:
Traceback (most recent call last):
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 270, in _print_async_exceptions
    future.result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 428, in result
    return self.__get_result()
  File "/usr/lib/python3.7/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 462, in async_fn
    self._write_async_result(id_vals, cols, i)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 94, in wrapped
    return method(self, *args, **kwds)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 421, in _write_async_result
    self._write(result)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 339, in _write
    self._write_fn(row, style)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/interface.py", line 354, in _write_update
    content, status, summary = self._content.update(row, style)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/common.py", line 858, in update
    content, status = super(ContentWithSummary, self).update(row, style)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/common.py", line 825, in update
    line, adjusted = self.fields.render(row, style)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/common.py", line 648, in render
    return self.style["separator_"].join(proc_fields) + "\n", adjusted
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/common.py", line 647, in <genexpr>
    proc_fields = (fld(val, keys=proc_keys) for fld, val in proc_fields)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/field.py", line 155, in __call__
    result = fn(value, result)
  File "/home/yoh/proj/dandi/dandi-cli/venv/dev3/lib/python3.7/site-packages/pyout/field.py", line 472, in split_flanks
    assert match, "This regexp should always match"
AssertionError: This regexp should always match
```

--- a/pyout/field.py
+++ b/pyout/field.py
@@ -456,7 +456,7 @@ class Flanks(object):
     """A pair of processors that split and rejoin flanking whitespace.
     """
 
-    flank_re = re.compile(r"(\s*)(.*\S)(\s*)\Z")
+    flank_re = re.compile(r"(\s*)(.*\S)(\s*)\Z", flags=re.DOTALL)
 
     def __init__(self):
         self.left, self.right = None, None

--- a/pyout/tests/test_tabular.py
+++ b/pyout/tests/test_tabular.py
@@ -367,6 +367,13 @@ def test_tabular_write_style_flanking():
     assert_eq_repr(out.stdout, expected)
 
 
+def test_tabular_write_style_flanking_newlines():
+    out = Tabular(columns=["name", "status"])
+    out({"name": "foo", "status": "a\nb"})
+    # The flanking regexp should match even if the field has new lines.
+    assert out.stdout.strip() == "foo a\nb"
+
+
 def test_tabular_write_align():
     out = Tabular(["name"],
                   style={"name": {"align": "right", "width": 10}})

